### PR TITLE
Update problem_format.md

### PR DESCRIPTION
**Pull request by 1-d**
Still have no idea how GitHub works

### CODE REVIEW

File: docs/judge/problem_format.md [hunk 1]
```diff
@@ -91,7 +86,7 @@ def check(proc_output, judge_output, judge_input, point_value, submission_source
 - `submission_source` contains the submission's source as a string. Useful for code golf problems.
 
 ## Returns
-A `CheckerResult` object (`from judge import CheckerResult`). A `CheckerResult` can be instantiated through `CheckerResult(case_passed_bool, points_awarded, partial_output='')`.
+A `CheckerResult` object (`from dmoj.result import CheckerResult`). A `CheckerResult` can be instantiated through `CheckerResult(case_passed_bool, points_awarded, partial_output='')`.
```
Comment (Xyene) [inline docs/judge/problem_format.md]: `partial_output` does not exist anymore; it's `feedback` now.

File: docs/judge/problem_format.md [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
 ## Batched Cases
-For batched cases, instead of `in` and `out`, the JSON object will contain `data` key, mapped to a list of objects, each containing `in` and `out`.
+The batch will contain the keys `points` and `batched`. `batched` will map to a list of batched cases, where each case contains `in` and `out`.
 
 # Interactive Problems - `grader`
-An `init.json` object can contain a top-level `grader` node, which contains a path to a Python file to be executed as a grader for the problem. The grader has access to the archive specified in `archive`. A `grader` Python script must implement a function that is called by the judge:
+An `init.yml` object can contain a top-level `grader` node, which contains a path to a Python file to be executed as a grader for the problem. The grader has access to the archive specified in `archive`. A `grader` Python script must implement a function that is called by the judge:
 
 ```python
 def grade(case, process, case_input=None, case_output=None, point_value=0, **kwargs)
```
Comment (BinAccel) [inline docs/judge/problem_format.md]: Signature is now `def grade(self, case):`

File: docs/judge/problem_format.md [hunk 3]
```diff
@@ -27,7 +27,7 @@ def grade(case, process, case_input=None, case_output=None, point_value=0, **kwa
 - `point_value` is an integer, the max points that can be awarded for the current testcase.
 
 ## Returns
-A `Result` object (`from judge import Result`). A result object has a `result_flag` field that stores a mask defining the current testcase result code. `proc_output` contains the string that will be displayed in the partial output pane.
+A `Result` object (`from dmoj.result import Result`). A result object has a `result_flag` field that stores a mask defining the current testcase result code. `proc_output` contains the string that will be displayed in the partial output pane.
 
 To illustrate, in a problem where the process must a line of echo input, an interactive approach would look like the following.
```
Comment (BinAccel) [inline docs/judge/problem_format.md]: Code sample below is outdated

File: docs/judge/problem_format.md [hunk 4]
```diff
@@ -56,26 +56,21 @@ def grade(case, process, case_input=None, case_output=None, point_value=0, **kwa
...
 
 # Custom Checkers - `checker`
-A problem with many possible outputs (e.g. not a single possible answer, with score based on accuracy) may benefit from the `checker` field in the `init.json` object. A checker is a Python script that is executed per-case, like an interactive grader, but which runs post-execution - it grades the output of a process but does not interact with it.
+A problem with many possible outputs (e.g. not a single possible answer, with score based on accuracy) may benefit from the `checker` field in the `init.yml` object. A checker is a Python script that is executed per-case, like an interactive grader, but which runs post-execution - it grades the output of a process but does not interact with it.
```
Comment (BinAccel) [inline docs/judge/problem_format.md]: Also mention that the global scope stays the same throughout grading.

File: docs/judge/problem_format.md (hunk 1 above)
Comment (1-d) [inline docs/judge/problem_format.md]: Fixed

File: docs/judge/problem_format.md [hunk 5]
```diff
@@ -16,11 +16,11 @@ The batch will contain the keys `points` and `batched`. `batched` will map to a
...
 
 ```python
-def grade(case, process, case_input=None, case_output=None, point_value=0, **kwargs)
+def grade(self, case):
 ```
 
 ## Parameters
-- `case` is an integer, the current test case with a zero-based index.
+- `case` is a `TestCase` object. `case.points` is the case's point value.
 - `process` is a `Popen`-like object with which the grader may interact. `process.stdin` is a file-like object representing the submission's `stdin` handle, and `process.stdout` wraps the process' `stdout`. Input is given to the process through `process.stdin` and output is typically read from `process.stdout`.
```
Comment (BinAccel) [inline docs/judge/problem_format.md]: You forgot something(s).

File: docs/judge/problem_format.md [hunk 6]
```diff
@@ -16,11 +16,11 @@ The batch will contain the keys `points` and `batched`. `batched` will map to a
 An `init.yml` object can contain a top-level `grader` node, which contains a path to a Python file to be executed as a grader for the problem. The grader has access to the archive specified in `archive`. A `grader` Python script must implement a function that is called by the judge:
 
 ```python
-def grade(case, process, case_input=None, case_output=None, point_value=0, **kwargs)
+def grade(self, case):
```
Comment (Xyene) [inline docs/judge/problem_format.md]: This is actually implemented by defining a `Grader` that takes `__init__(judge, problem, language, source)`. This `Grader` typically inherits `StandardGrader` (https://github.com/DMOJ/judge/blob/master/dmoj/graders/standard.py)

File: docs/judge/problem_format.md [hunk 7]
```diff
@@ -75,7 +75,7 @@ A problem with many possible outputs (e.g. not a single possible answer, with sc
 A `checker` Python script must implement a function that is called by the judge:
 
 ```python
-def check(proc_output, judge_output, judge_input, point_value, submission_source, **kwargs)
+def check(proc_output, judge_output, judge_input, point_value, submission_source, **kwargs):
```
Comment (BinAccel) [inline docs/judge/problem_format.md]: `judge_input`, `point_value`, and `submission_source` are all part of `kwargs` now; it should be `def check(proc_output, judge_output, **kwargs):`

Comment (Xyene) [inline docs/judge/problem_format.md]: Amusingly enough it's always been that way, except we were passing them (and probably still are) in such a way that that format was OK.

Comment (BinAccel) [inline docs/judge/problem_format.md]: Might as well update it so lazy people ~like me~ can continue to be lazy.

File: docs/judge/problem_format.md (hunk 5 above)
Comment (1-d) [inline docs/judge/problem_format.md]: Should be better now, except for `process`

File: docs/judge/problem_format.md (hunk 2 above)
Comment (1-d) [inline docs/judge/problem_format.md]: Fixed

File: docs/judge/problem_format.md (hunk 4 above)
Comment (1-d) [inline docs/judge/problem_format.md]: Done

File: docs/judge/problem_format.md (hunk 7 above)
Comment (1-d) [inline docs/judge/problem_format.md]: Fixed, I guess

File: docs/judge/problem_format.md [hunk 8]
```diff
@@ -1,19 +1,19 @@
-Each problem is stored in its own directory. That directory must contain a file named `init.json`.
+Each problem is stored in its own directory. That directory must contain a file named `init.yml`.
 
-# `init.json`
-The entire file is a JSON Object. It must contain one key, `test_cases`, which maps to a list of test cases. Optionally, but almost always, will there be an `archive` key, which allows the problem data to be stored, compressed, in a `.zip` file, instead of the problem directory as flat files.
+# `init.yml`
+The entire file is a YAML object. It must contain one key, `test_cases`, which maps to a list of test cases. Optionally, but almost always, will there be an `archive` key, which allows the problem data to be stored, compressed, in a `.zip` file, instead of the problem directory as flat files.
```
Comment (BinAccel) [inline docs/judge/problem_format.md]: `key` => `value indicator`

File: docs/judge/problem_format.md [hunk 9]
```diff
@@ -1,19 +1,19 @@
...
 
 # `test_cases`
-Each element in the `test_cases` list is a JSON object. It contains the key `points`, mapped to an integer specifying the number of points that test case is worth.
+Each element in the `test_cases` list is a YAML object. It contains the key `points`, mapped to an integer specifying the number of points that test case is worth.
```
Comment (BinAccel) [inline docs/judge/problem_format.md]: `YAML object` => `YAML keyed branch`

File: docs/judge/problem_format.md (hunk 8 above)
Comment (1-d) [inline docs/judge/problem_format.md]: In YAML with `a: b`, I'm pretty sure that `a` is the key, `:` is the value indicator, and `b` is the value

File: docs/judge/problem_format.md (hunk 9 above)
Comment (1-d) [inline docs/judge/problem_format.md]: I think this is a little specific, but OK

File: docs/judge/problem_format.md (hunk 8 above)
Comment (BinAccel) [inline docs/judge/problem_format.md]: My bad.

**Comment by 1-d** [comment]
Added a generic YAML example

**Pull request merged**